### PR TITLE
ENH: Add config for pre-commit

### PR DESCRIPTION
**Pull request by charlesdong1991**
this is just a draft which kind of shows what i wanna say for #755 

i set it to WIP, so feel free to comment and i will change. @ueshin @HyukjinKwon

### CODE REVIEW

**Comment by charlesdong1991** [comment]
ahh, just checked `.travis.yml`, and find out it's building py35, but `black` only runs on py36 or newer, so pip install will fail..

**Comment by ueshin** [comment]
Maybe we don't need to add it as a dependency, but in that case, definitely we have to add some instruction to install the library for contributors.

**Comment by charlesdong1991** [comment]
yeah, agree... have to remove it, i will add some instructions in `CONTRIBUTING.md`.

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  | \.mypy_cache
+  | \.tox
+  | \.venv
+  | _build
+  | buck-out
+  | build
+  | dist
+  | venv
+)/
+'''
```
Comment (ueshin) [inline pyproject.toml]: nit: add a line break at the end of a file.

Comment (charlesdong1991) [reply pyproject.toml]: thanks, added!
i also notice line length used in koalas is 100, not 88 when reading through `contributing.md`, also changed in this file.

**Comment by HyukjinKwon** [comment]
To me, I'm okie. Seems worth trying.

**Pull request closed without merging**